plugins/destination/plugin_testing.go

```diff
@@ -101,10 +101,24 @@ func getTestLogger(t *testing.T) zerolog.Logger {
 type NewPluginFunc func() *Plugin
 
 type PluginTestSuiteRunnerOptions struct {
-	IgnoreNullsInLists bool // strip nulls from lists before checking equality. Destination setups that don't support nulls in lists should set this to true.
+	// IgnoreNullsInLists allows stripping null values from lists before comparison.
+	// Destination setups that don't support nulls in lists should set this to true.
+	IgnoreNullsInLists bool
+
+	// AllowNull is a custom func to determine whether a data type may be correctly represented as null.
+	// Destinations that have problems representing some data types should provide a custom implementation here.
+	// If this param is empty, the default is to allow all data types to be nullable.
+	AllowNull func(arrow.DataType) bool
```

Comment:
Destinations don't really control the schema of the data they receive; what will happen if a destination that doesn't support nulls receives a nullable schema? How can we be sure if we're not testing it? I think we should still be testing that case, but we can be more flexible with the comparisons (like in `IgnoreNullsInLists` above). WDYT?

Reply:
Well, that's true, but it's also tru for the `skip maps` & other skip options for test tables.

IMO, we should instead base what types are available by the available scalar implementations.

As for this one, it's just a work-around for tests.
I think another option could be allowing treating empty values as null ones for specific types, but then we won't be able to use Arrow record approx equal and would instead need to traverse the record ourselves.

Reply:
> I think another option could be allowing treating empty values as null ones for specific types, but then we won't be able to use Arrow record approx equal and would instead need to traverse the record ourselves.

Yes, I think this would be a better solution. Like in the `IgnoreNullsInLists` case, we still want to test the plugins with these constraints, but we need to adapt the test expectations for them. And like in `IgnoreNullsInLists`, I think we should still be able to use record approx equal if we apply a transformation to the relevant types first (e.g. converting empty values to null, or whatever the case needs to be).